### ISSUE TITLE
Eagerly calling a Keras Conv2D crashes with `floating point exception`

### ISSUE THREAD

**Issue by psybuzz** [tensorflow/tensorflow#48094]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): MacOS 11.2
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.5.0.dev20210215
- Python version: 3.8.2
- CUDA/cuDNN version: n/a
- GPU model and memory: AMD Radeon Pro 5500M 8 GB

**Standalone code to reproduce the issue**

In a virtualenv with TensorFlow installed, run

```python
import tensorflow as tf
tf.keras.layers.Conv2D(100, 3)(tf.constant([[[[]]]]))
```

**Describe the current behavior**

Entire process dies, printing this error to the terminal: `floating point exception  python`

**Describe the expected behavior**

Process does not die.  I would probably expect a `ValueError` or similar exception.

**Comment by Saduf2019** [comment]
I ran the code shared but colab crashes, please find the [gist here](https://colab.research.google.com/gist/Saduf2019/73e4368b893d9089f7eb41408218ab14/untitled575.ipynb)

**Comment by AdityaKane2001** [comment]
@psybuzz 
The code given as the example at [keras website](https://keras.io/api/layers/convolution_layers/convolution2d/) works as expected. Gist [here](https://colab.research.google.com/gist/AdityaKane2001/b3fc660078d46974db4797052b6f3256/untitled575.ipynb). Also,
`a = tf.keras.layers.Conv2D(10, (3,3),activation = 'relu')(tf.constant(np.random.rand(10,10,10,3)))`
this works as expected. 
> tf.keras.layers.Conv2D(100, 3)(tf.constant([[[[]]]]))

As you mentioned, this crashes the session.

**Comment by ymodak** [comment]
This is fixed with TF 2.5 It raises an error and avoids crashing
```python
InvalidArgumentError: filter depth must be stricly positive, got 0 [Op:Conv2D]
```

**Comment by psybuzz** [comment]
Thanks, I checked with TF 2.5.0 and can confirm it throws an error!
